CaptureAgentsEndpointTest requires unused network port

Running the admin interface tests failed with:

```
[INFO] Running org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.167 s <<< FAILURE! - in org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest
Error:  org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest  Time elapsed: 0.167 s  <<< ERROR!
java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8084
	at org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest.oneTimeSetUp(CaptureAgentsEndpointTest.java:193)
Caused by: java.net.BindException: Address already in use
	at org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest.oneTimeSetUp(CaptureAgentsEndpointTest.java:193)
```

…which is most likely a concurrency problem.
I any case, the tests shouldn't hard-code a single port but at least try a range of ports if one is already in use.

It seems like something doesn't seem to work like it should. I've seen at least two more exceptions like this:

```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.141 s <<< FAILURE! - in org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest
Error:  org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest  Time elapsed: 0.14 s  <<< ERROR!
java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8084
	at org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest.oneTimeSetUp(CaptureAgentsEndpointTest.java:193)
Caused by: java.net.BindException: Address already in use
	at org.opencastproject.adminui.endpoint.CaptureAgentsEndpointTest.oneTimeSetUp(CaptureAgentsEndpointTest.java:193)
```

Every time it seems like the test is trying to use port 8084. That's less random than expected…

--- a/modules/rest-test-environment/src/main/java/org/opencastproject/test/rest/RestServiceTestEnv.java
+++ b/modules/rest-test-environment/src/main/java/org/opencastproject/test/rest/RestServiceTestEnv.java
@@ -38,7 +38,7 @@ import org.slf4j.LoggerFactory;
 import java.io.IOException;
 import java.net.URL;
 import java.net.URLConnection;
-import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * Helper environment for creating REST service unit tests.
@@ -115,16 +115,21 @@ public final class RestServiceTestEnv {
    * Return a localhost base URL with a random port between 8081 and 9000. The method features a port usage detection to
    * ensure it returns a free port.
    */
-  public static URL localhostRandomPort() {
+  public static synchronized URL localhostRandomPort() {
     for (int tries = 100; tries > 0; tries--) {
-      final URL url = UrlSupport.url("http", "localhost", 8081 + new Random(System.currentTimeMillis()).nextInt(919));
+      final int random = ThreadLocalRandom.current().nextInt(62000);
+      final URL url = UrlSupport.url("http", "127.0.0.1", 3000 + random);
       try {
         final URLConnection con = url.openConnection();
         con.setConnectTimeout(1000);
         con.setReadTimeout(1000);
         con.getInputStream();
+        Thread.sleep(100);
       } catch (IOException e) {
+        logger.debug("Selected URL: {}", url);
         return url;
+      } catch (InterruptedException e) {
+        // ignore sleep interruption
       }
     }
     throw new RuntimeException("Cannot find free port. Giving up.");

